src/main/java/Storage.java

```diff
@@ -0,0 +1,114 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * @author bakwxh
+ * @version 0.1
+ */
+public class Storage {
+	/**
+	 * Path to save file.
+	 */
+	private String filePath;
+
+	/**
+	 * Constructor.
+	 * @param filePath Path to save file.
+	 */
+	public Storage(final String filePath) {
+		this.filePath = filePath;
+	}
+
+	/**
+	 * Saves memory to file.
+	 * @param tasks Task list.
+	 * @throws DukeException Exceptions.
+	 */
+	public void saveMemory(final TaskList tasks) throws DukeException {
+		ArrayList<Task> memory = tasks.getMemory();
+		String dir = System.getProperty("user.dir") + "/savedData.txt";
+		BufferedWriter out;
+		try {
+			out = new BufferedWriter(new FileWriter(dir));
+		} catch (IOException e1) {
+			throw new DukeException("Error saving data.");
+		}
+		for (Task t : memory) {
+			try {
+				out.write(t.toSave() + "\n");
+			} catch (IOException e) {
+				throw new DukeException("Error saving data.");
+			}
+		}
+		try {
+			out.flush();
+			out.close();
+		} catch (IOException e) {
+			throw new DukeException("Error saving data.");
+		}
+	}
+
+	/**
+	 * Loads from save file.
+	 * @return Returns array list with saved tasks.
+	 * @throws DukeException Exceptions.
+	 */
+	public ArrayList<Task> load() throws DukeException {
+		ArrayList<Task> taskList = new ArrayList<>();
+		try {
+			FileInputStream test = new FileInputStream(filePath);
+			Scanner sc = new Scanner(test);
+			while (sc.hasNextLine()) {
+				String rawInput = sc.nextLine();
+				if (rawInput.toLowerCase().startsWith("done")) {
```

Comment:
Felt rather verbose here. Are there ways to make this more succinct and concise?